src/v2/client/startLoginFlow.js

```diff
@@ -38,13 +38,21 @@ export async function startLoginFlow(settings) {
   if (useInteractionHandle) {
     return interact(settings);
   }
-
   // Use stateToken from session storage if exists
   // See more details at ./docs/use-session-token-prior-to-settings.png
   const stateHandleFromSession = sessionStorageHelper.getStateHandle();
+
   if (stateHandleFromSession) {
     return introspect(settings, stateHandleFromSession)
       .then((idxResp) => {
+        const oldAppId = sessionStorageHelper.getAppId();
+        const newAppId = idxResp?.context?.app?.value?.id;
+
+        if (oldAppId && (oldAppId !== newAppId)) {
+          sessionStorageHelper.removeStateHandle();
+          return startLoginFlow(settings);
+        }
+
         // 1. abandon the settings.stateHandle given session.stateHandle is still valid
```

Comment:
nit: (as unrelated to your changes)
The comment need expand a little bit to explain why `session.stateHandle is still valid` when hit this line.
I'm thinking
```
abandon the settings.stateHandle given session.stateHandle is still valid otherwise the flow goes to 'catch' block.
```

Reply:
If `introspect` call succeeded when using stored staetToken `sessionStorageHelper.getStateHandle();` then it is still valid.